bokehjs/src/coffee/api/typings/models/sources.d.ts

```diff
@@ -5,10 +5,13 @@ declare namespace Bokeh {
         callback: Callback;
     }
 
+    type Data = {[key: string]: ArrayLike<any>};
+
     export var ColumnDataSource: { new(attributes?: KeyVal, options?: KeyVal): ColumnDataSource };
     export interface ColumnDataSource extends DataSource {
-        data: {[key: string]: ArrayLike<any>};
-        stream(new_data: Object, rollover: Number): void;
+        data: Data;
+
+        stream(new_data: Data, rollover: number): void;
```

Comment:
It's actually a member and a function, and there might be other in future. This is static typing, i.e. you want to be as precise as possible. Note that this typing isn't that specific, because it still uses `any`. This could have been replaced naively with `number | string`, but that might be insufficient. Unfortunately typescript doesn't have tools to express the correct type, because you need an open (typeclass-based, like in Haskell) type system.